lib/cybersource/api.rb

```diff
@@ -276,6 +279,20 @@ def should_credit?(kb_payment_id, context, options = {})
         (now - transaction.created_at) >= threshold
       end
 
+      def before_gateways(kb_transaction, last_transaction, payment_source, amount_in_cents, currency, options, context = nil)
+        # Provide necessary information for merchant descriptor
+        if options[:merchant_descriptor].present? && options[:merchant_descriptor]['is_amex'].nil?
```

Comment:
Hi @pierre ,
This is the most difficult part of this PR, for our use case, the credit card information will only passed through the authorize call (actually the combo call), and no card information will be in the capture, refund and void call. In this case, I think the cc_type will be empty in the transaction requests other than authorize. If this is true, then I think check `cc_type` is not worthy.
For `respond_to?(:brand)`, will update the code in the next commit.